### DISCUSSION TITLE
How to update one to many relations

### DISCUSSION THREAD

**Discussion by kane1997** [prisma/prisma#15961] [Q&A] [answered]
my req.body is
`
{ "remark": "et esse qui", "enabled": false, "name": "dedwcdw", "brandId": 2, "variants": [ { "sku": "reprehenderit", "ean": "aliquip occaecat qui aute voluptate", "remark": "dolore Duis nostrud pariatur ex", "enabled": true, "name": "sdcdcds" }, { "sku": "laborum ut aute mollit", "ean": "in ullamco ut", "remark": "velit qui dolore", "enabled": false, "name": "dsxcsdcsd" }, { "ean": "enim deserunt minim fugiat incididunt", "name": "sdcdcdscsd", "sku": "aute", "enabled": true, "remark": "id nulla" } ] }`

and my shcema is

`
model Brand {
id Int @id @unique @default(autoincrement()) @Map("brand_id")
name String? @unique @Map("brand_name")
enabled Boolean? @default(true) @Map("brand_enabled")
remark String? @default("") @Map("brand_remark")
created_at DateTime @default(now()) @Map("brand_created_at")
updated_at DateTime? @updatedAt @Map("brand_updated_at")

products Product[]

@@Map("t_brand")
}

model Product {
id Int @id @unique @default(autoincrement()) @Map("product_id")
name String? @unique @Map("product_name")
remark String? @default("") @Map("product_remark")
enabled Boolean? @default(true) @Map("product_is_enabled")
created_at DateTime @default(now()) @Map("product_created_at")
updated_at DateTime? @updatedAt @Map("product_updated_at")
product_created_by User? @relation(name: "product_created_by", fields: [created_by], references: [id], onDelete: Cascade, onUpdate: Cascade)
created_by Int?
product_updated_by User? @relation(name: "product_updated_by", fields: [updated_by], references: [id], onDelete: Cascade, onUpdate: Cascade)
updated_by Int?
brands Brand? @relation(fields: [brandId], references: [id])
brandId Int?

product_variants Product_Variant[]

@@Map("t_product")
}

model Product_Variant {
id Int @id @/unique @default(autoincrement()) @Map("product_variant_id")
name String? @Map("product_variant_name")
sku String? @Map("product_variant_sku")
ean String? @Map("product_variant_ean")
enabled Boolean? @default(false) @Map("product_variant_is_enabled")
remark String @Map("product_variant_remark")
created_at DateTime @default(now()) @Map("product_variant_created_at")
updated_at DateTime? @updatedAt @Map("product_variant_updated_at")
created_By User? @relation(name: "product_variant_created_by", fields: [created_by], references: [id], onDelete: Cascade, onUpdate: Cascade)
created_by Int?
updated_By User? @relation(name: "product_variant_updated_by", fields: [updated_by], references: [id], onDelete: Cascade, onUpdate: Cascade)
updated_by Int?

products Product? @relation(fields: [productId], references: [id])
productId Int?

promotion_plans Promotion_Plan[]

@@Map("t_product_variant")
}
`

my script : is

`const update = await prisma.product.update({ where: { id: Number(productId), }, data: { name: name, remark: remark, enabled: enabled, brandId: brandId, product_variants: { update: variants, }, }, }); can anyone help me how to update many variants through one request here?`

### can anyone help me how to update many variants through one request here?

**Comment by nurul3101** [comment, marked as answer]
Answered in this Discussion: #15959

**Comment by Mae6e** [comment]
update one-many relationship data with prisma

    const response = await prisma.currency.update({
        where: { id: data.id },
        data: {
            ...data,
            currencyInNetworks: {
                updateMany: toEdit.map(network => ({
                    where: { id: network.id },
                    data: network
                })),
                create: toCreate,
                deleteMany: {
                    id: { in: toDelete }
                }
            }
        }
    });

**Reply by neilhuyton** [reply]
Thank you. Best post on the internet.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
